framework/Util/Cron/TCronTask.php

```diff
@@ -0,0 +1,247 @@
+	abstract public function execute($cronModule);
+	
+	/**
+	 * @return string the unique name of the Task
+	 */
+	public function getName()
+	{
+		return $this->_name;
+	}
+
+	/**
+	 * @param string $name the unique name of the Task
+	 */
+	public function setName($name)
+	{
+		$this->_name = TPropertyValue::ensureString($name);
+	}
+	
+	/**
+	 * @return string the cron style schedule of the task
+	 */
+	public function getSchedule()
+	{
+		return $this->_schedule;
+	}
+	
+	/**
+	 *
+	 * @param string $schedule the cron style schedule of the task
+	 */
+	public function setSchedule($schedule)
+	{
+		$this->_schedule = TPropertyValue::ensureString($schedule);
+		if ($this->_scheduler) {
+			$this->_scheduler->setSchedule($this->_schedule);
+		}
+	}
+	
+	/**
+	 * @return string the task being executed
+	 */
+	public function getTask()
+	{
+		return get_class($this);
+	}
+	
+	/**
+	 * @return string the user id executing the Task
+	 */
+	public function getUserId()
+	{
+		return $this->_userId;
+	}
+
+	/**
+	 * @param string $userid the user id executing the Task
+	 */
+	public function setUserId($userid)
+	{
+		$this->_userId = TPropertyValue::ensureString($userid);
+	}
+	
+	/**
+	 * @return null|string the utility module for the task
+	 */
+	public function getModuleId()
+	{
+		return $this->_moduleId;
+	}
+	
+	/**
+	 *
+	 * @param string $moduleId the utility module for the task.
+	 */
+	public function setModuleId($moduleId)
+	{
+		$this->_moduleId = ($moduleId !== null) ? TPropertyValue::ensureString($moduleId) : null;
+	}
+	
+	/**
+	 * @return null|Prado\IModule returns the module from ModuleId
+	 */
+	public function getModule()
+	{
+		$app = $this->getApplication();
+		$module = $app->getModule($this->_moduleId);
+		if ($module === null && $this->_moduleId !== null) {
+			throw new TConfigurationException('crontask_no_module', $this->_moduleId);
+		}
+		return $module;
+	}
+	
+	/**
+	 * @return int the number of times the task has run
+	 */
+	public function getProcessCount()
+	{
+		return $this->_processCount;
+	}
+	
+	/**
+	 *
+	 * @param int $count the number of times the task has run
+	 */
+	public function setProcessCount($count)
+	{
+		return $this->_processCount = $count;
+	}
+	
+	/**
+	 * @return numeric the time of running this cron task
+	 */
+	public function getLastExecTime()
+	{
+		return $this->_lastexectime;
+	}
+	
+	/**
+	 * sometimes floats don't output correctly to 6 significant figures (microtime).
+	 * @param numeric $v the time of running this cron task
+	 */
+	public function setLastExecTime($v)
+	{
+		$this->_lastexectime = TPropertyValue::ensureInteger($v);
+		//$this->_lastexectime = rtrim(rtrim(number_format($v, 6,'.',''), '0'), '.');
+	}
+	
+	/**
+	 * @return numeric time of the next trigger after the lastExecTime
+	 */
+	public function getNextTriggerTime()
+	{
+		$s = $this->getScheduler();
+		return $s->getNextTriggerTime($this->_lastexectime);
+	}
+	
+	/**
+	 *
+	 */
+	/*public function getPreviousTriggerTime()
```

Comment:
Is this a leftover to be removed or you want to decomment and keep it?

Reply:
I wanted to implement it, but that is a LOT of work just to get the NextTriggerTime working.  phew!   took me a week and a half just to get the Next working.  The previous is optional..  so, I'll remove it for now.

Reply:
If you want to implement it, please, by all means, go for it.  It should be simply duplicated code from the NextTriggerTime except backwards.   The test units are the real monster.

Reply:
I don't see a point for it in a simple cron scheduler. Feel free to drop it